saleor/discount/utils.py

```diff
@@ -345,17 +405,77 @@ def create_or_update_discount_objects_from_promotion_for_checkout(
     lines_info: Iterable["CheckoutLineInfo"],
 ):
     create_discount_objects_for_catalogue_promotions(lines_info)
-    create_discount_objects_for_order_promotions(checkout_info, lines_info)
+    create_checkout_discount_objects_for_order_promotions(checkout_info, lines_info)
+
+
+def create_or_update_discount_objects_from_promotion_for_order(
+    order: "Order",
+    lines_info: Iterable["DraftOrderLineInfo"],
+):
+    create_discount_objects_for_catalogue_promotions(lines_info)
+    _update_order_line_prefetched_discounts(lines_info)
+    _update_order_line_base_unit_prices(lines_info)
```

Comment:
IMO we shouldn't update the prices here. We just need the proper values on `order.subtotal` and `order.total`. Seems like we can set the proper values in `_set_order_base_prices` function with use of `lines_info`

Reply:
You are right:) I moved the logic to `_set_order_base_prices functions` function

Reply:
We need to update `base_unit_price` in order to calculate correct subtotal in `_set_order_base_prices`.
I would keep it separately, because `_update_order_line_base_unit_prices` refresh prices after applying catalogue promotions (so it should be a part of `create_order_line_discount_objects_for_catalogue_promotions`), while `_set_order_base_prices` make calculations for applying order promotion.